src/main/kotlin/no/nav/syfo/cronjob/narmesteLederRevarsel/ResendNarmesteLederVarselCronjob.kt

```diff
@@ -0,0 +1,65 @@
+package no.nav.syfo.cronjob.narmesteLederRevarsel
+
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.brev.narmesteleder.NarmesteLederVarselService
+import no.nav.syfo.client.narmesteleder.NarmesteLederClient
+import no.nav.syfo.cronjob.DialogmoteCronjob
+import no.nav.syfo.cronjob.DialogmoteCronjobResult
+import no.nav.syfo.dialogmote.database.getMotedeltakerArbeidsgiverVarsel
+import no.nav.syfo.dialogmote.domain.MotedeltakerVarselType
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.domain.Virksomhetsnummer
+import org.slf4j.LoggerFactory
+import java.time.LocalDateTime
+
+class ResendNarmesteLederVarselCronjob(
+    private val narmesteLederClient: NarmesteLederClient,
+    private val narmesteLederVarselService: NarmesteLederVarselService,
+    private val database: DatabaseInterface,
+) : DialogmoteCronjob {
+    override val initialDelayMinutes: Long = 2
+    override val intervalDelayMinutes: Long = 9999999
+
+    override suspend fun run() {
+        val resendResult = DialogmoteCronjobResult()
+
+        resendVarsler(resendResult)
+
+        log.info(
+            "RESEND-TRACE: Completed resending of varsler with result: {}, {}",
+            StructuredArguments.keyValue("failed", resendResult.failed),
+            StructuredArguments.keyValue("updated", resendResult.updated),
+        )
+    }
+
+    private suspend fun resendVarsler(
+        resendResult: DialogmoteCronjobResult
+    ) {
+        val startDateTime = LocalDateTime.of(2022, 9, 12, 22, 10)
+        val endDateTime = LocalDateTime.of(2022, 9, 13, 15, 15)
+        val varslerToResend = database.getMotedeltakerArbeidsgiverVarsel(
+            startDateTime = startDateTime,
+            endDateTime = endDateTime,
+        )
+
+        log.info("RESEND-TRACE: Found ${varslerToResend.size} varsler.")
+
+        varslerToResend.forEach { varsel ->
+            try {
+                val narmesteLeder = narmesteLederClient.activeLederSystemToken(
+                    personIdentNumber = PersonIdentNumber(varsel.second),
+                    virksomhetsnummer = Virksomhetsnummer(varsel.third),
+                )
+                narmesteLederVarselService.sendVarsel(narmesteLeder!!, MotedeltakerVarselType.valueOf(varsel.first), false)
```

Comment:
Dersomatte, hvisomatte `narmesteLeder` er `null`, hva skjer da? Vil det ødelegge for sending av de andre varslene?

Reply:
Da skal `narmesteLeder!!` kaste en NullPointer, og så skal det catches og logges før vi går videre 👍

Reply:
Fikser dette nå, så vi kun logger uten å kaste exception 🙂